src/features/groups/add-group/service-accounts-list.tsx

```diff
@@ -28,105 +37,186 @@ const reducer = ({ serviceAccountReducer }: { serviceAccountReducer: ServiceAcco
+    value === BulkSelectValue.all && onSelect(true, selectableRows);
+    value === BulkSelectValue.nonePage && onSelect(false, rows);
+    value === BulkSelectValue.page && onSelect(true, selectableRows);
+  };
+
+  const pageSelected = rows.length > 0 && rows.every(isSelected);
+  const pagePartiallySelected = !pageSelected && rows.some(isSelected);
+  const totalCount = status === LAST_PAGE ? offset + serviceAccounts.length : offset + serviceAccounts.length + 1;
+
+  const ouiaId = 'group-add-service-accounts';
+
   return (
-    <TableToolbarView
-      className="rbac-service-accounts-list"
-      columns={columns}
-      isSelectable
-      rows={createRows(serviceAccounts, selected)}
-      data={serviceAccounts}
-      fetchData={fetchAccounts}
-      isLoading={isLoading}
-      pagination={{
-        limit,
-        offset,
-        ...(status === LAST_PAGE ? { count: offset + serviceAccounts.length } : {}),
-      }}
-      paginationProps={{
-        toggleTemplate: () => (
-          <>
-            <b>
-              {offset + 1} - {offset + serviceAccounts.length}
-            </b>{' '}
-            of <b>{status === LAST_PAGE ? offset + serviceAccounts.length : 'many'}</b>
-          </>
-        ),
-        isCompact: true,
-      }}
-      checkedRows={selected}
-      setCheckedItems={setSelected}
-      titlePlural={intl.formatMessage(messages.serviceAccounts).toLowerCase()}
-      titleSingular={intl.formatMessage(messages.serviceAccount)}
-      emptyProps={{
-        title: intl.formatMessage(messages.noServiceAccountsFound),
-        description: [intl.formatMessage(messages.contactServiceTeamForAccounts), ''],
-      }}
-      tableId="group-add-accounts"
-      ouiaId="group-add-accounts"
-      toolbarButtons={() => []}
-      routes={() => null}
-      setFilterValue={() => {}}
-      filters={[]}
-    />
+    <div className="rbac-service-accounts-list">
+      <DataView ouiaId={ouiaId} selection={selection} activeState={activeState}>
+        <DataViewToolbar
+          ouiaId={`${ouiaId}-header-toolbar`}
+          bulkSelect={
+            <BulkSelect
+              isDataPaginated
+              pageCount={serviceAccounts.length}
+              totalCount={totalCount}
+              selectedCount={dataViewSelected.length}
+              pageSelected={pageSelected}
+              pagePartiallySelected={pagePartiallySelected}
+              onSelect={handleBulkSelect}
+            />
+          }
+          pagination={
+            <Pagination
+              perPageOptions={PER_PAGE_OPTIONS}
+              itemCount={totalCount}
+              page={page}
+              perPage={perPage}
+              onSetPage={onSetPage}
+              onPerPageSelect={onPerPageSelect}
+            />
+          }
+        />
+        {isLoading ? (
+          <SkeletonTable rowsCount={10} columns={columns} variant={TableVariant.compact} />
+        ) : (
+          <DataViewTable
+            columns={columns}
```

Comment:
Let's use compact table as well
```suggestion
           variant="compact"
            columns={columns}
```